### PR TITLE
Add Chaos Engineering and Testing ADR

### PR DESCRIPTION
**Pull request by chaitanyaenr**
This commit adds an ADR which walks through:
- Current problem around possibility ofa system or service not designed and
  architected to handle failures leading to downtime in production environment.
- Proposed solutions to overcome the problem by publishing best practices,
  methodology and test framework to use to drive adoption of chaos testing.

### CODE REVIEW

**Comment by chaitanyaenr** [comment]
@david-martin PTAL. Thanks.

File: ADRs/RH/SIG-SRE/ADR-00002-Chaos-Testing-and-Engineering.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+There are assumptions that users might have when operating and running their applications in distributed systems environment:
+
+The network is reliable.
+There is consistent resource usage with no spikes.
+There is zero latency.
+Bandwidth is infinite.
+The network is secure.
+Topology never changes.
+The network is homogeneous.
```
Comment (david-martin) [inline ADRs/RH/SIG-SRE/ADR-00002-Chaos-Testing-and-Engineering.md]:
```suggestion
* The network is reliable.
* There is consistent resource usage with no spikes.
* There is zero latency.
* Bandwidth is infinite.
* The network is secure.
* Topology never changes.
* The network is homogeneous.

```

**Review by david-martin — COMMENTED** [review]
1 suggestion, otherwise good to merge

File: ADRs/RH/SIG-SRE/ADR-00002-Chaos-Testing-and-Engineering.md (hunk 1 above)
Comment (david-martin) [reply ADRs/RH/SIG-SRE/ADR-00002-Chaos-Testing-and-Engineering.md]: The preview showed this as a single paragraph, so converting to a list for easier reading and emphasis on the number of assumptions.

**Comment by chaitanyaenr** [comment]
> 1 suggestion, otherwise good to merge

Fixed. Thanks!

**Review by lisa — DISMISSED** [review]
/hold

As mentioned in #7, there is an ADR numbering conflict between #7, #11 and #12 (this PR). Each of them seems to want to use ADR-00002. Do you want to work with the other authors to resolve the conflict?

**Comment by chaitanyaenr** [comment]
@lisa @david-martin I will bump up the ADR number of this draft to ADR-00004, this way #7 and #11 can use ADR-00002 and ADR-00003. Thanks.

**Comment by lisa** [comment]
@chaitanyaenr Lovely, thank you.

**Comment by chaitanyaenr** [comment]
> @lisa @david-martin I will bump up the ADR number of this draft to ADR-00004, this way #7 and #11 can use ADR-00002 and ADR-00003. Thanks.

Done!

**Pull request merged**